backend/station_manager/app/src/main/java/com/StationManager/app/storage/repository/IRepository.java

```diff
@@ -0,0 +1,12 @@
+package com.StationManager.app.storage.repository;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface IRepository<T> {
+    void add(T entity);
+    void update(T entity);
+    void remove(T entity);
+    Optional<T> getById(int id);
```

Comment:
not every entity will have an identifier, so I would remove it.
Either way, arguably though, not every entity would be updatable